sdk/python/kfp/dsl/__init__.py

```diff
@@ -247,3 +224,32 @@ def my_pipeline():
         producer_task = artifact_producer()
         artifact_consumer(model=producer_task.output)
 """
+
+# compile-time only dependencies
+if os.environ.get('_RUNTIME', 'false') != 'true':
```

Comment:
nit: although the chance of collision should be very low, consider using a more specific name, e.g.: `KFP_RUNTIME`

Reply:
Good thinking. Done. I used `_KFP_RUNTIME` with the hope that the leading prefix signals to Python developers that this variable is private. Ideally this is an implementation detail of the runtime code that no user code depends on.